src/main/java/duke/commands/CommandDeadlineHandler.java

```diff
@@ -0,0 +1,44 @@
+package duke.commands;
+
+import java.time.format.DateTimeParseException;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.utils.DateTime;
+
+/**
+ * Handler for the Deadline command.
+ */
+public class CommandDeadlineHandler extends CommandHandler {
+
+    public CommandDeadlineHandler(String value, String flag, String additionalValue) throws DukeException {
+        super(value, flag, additionalValue);
+    }
+
+    /**
+     * Checks the validity of the syntax after being parsed by CommandParser.
+     * @throws DukeException error message.
+     */
+    public void checkValid() throws DukeException {
+        if (value == null || !flag.equals("by") || additionalValue == null) {
+            throw new DukeException("Correct usage: deadline return book /by 24/04/2019 1600");
```

Comment:
I like the way you throw an exception and help the user to type in the correct command at the same time.